Initialize Bond TP

shannynalayna opened the pull request:
Initializes Bond transaction processor by instantiating a connection
with the sawtooth validator. Also initializes the transaction handler
with metadata, further business logic is yet to be implemented.

Also adds a function for specifically creating and getting the bond 
namespace to allow for greater abstraction. Modifies the bond processor
to utilize this function.

Signed-off-by: Shannyn Telander <telander@bitwise.io>

Commit:
Initialize Bond TP

Initializes Bond transaction processor by instantiating a connection
with the sawtooth validator. Also initializes the transaction handler
with metadata, further business logic is yet to be implemented.

Signed-off-by: Shannyn Telander <telander@bitwise.io>

dplumb94 commented on processor/src/handler.rs:
```diff
@@ -0,0 +1,60 @@
+    fn apply(
+        &self,
+        request: &TpProcessRequest,
+        context: &mut TransactionContext,
```
Got a couple compiler warnings here: 
```
warning: unused variable: `request`
  --> src/handler.rs:55:9
   |
55 |         request: &TpProcessRequest,
   |         ^^^^^^^ help: consider using `_request` instead
   |
   = note: #[warn(unused_variables)] on by default

warning: unused variable: `context`
  --> src/handler.rs:56:9
   |
56 |         context: &mut TransactionContext,
   |         ^^^^^^^ help: consider using `_context` instead
```
I'd recommend using the suggested underscores for now, and then change them back when you are actually using the variables. That way we don't get warnings during the build process and the code is still correct.

dplumb94 commented on processor/Cargo.toml:
```diff
@@ -19,3 +19,11 @@ authors = ["Bitwise IO"]
 bond-common = { path = "../common"}
+protobuf = "2"
+rust-crypto = "0.2.36"
+zmq = { git = "https://github.com/erickt/rust-zmq", branch = "release/v0.8" }
```
Do you get any issues if you omit this dependency? The crate is not being directly used in Bond, so I don't think you need it here. It is used by the Sawtooth SDK, which is why you need to `apt-get install` the `libzmq3-dev` library in the container (that library is a prereq for this zmq wrapper crate).

shannynalayna commented:
@dplumb94 updated

eloaverona commented on processor/src/handler.rs:
```diff
@@ -0,0 +1,60 @@
+
+use bond_common::addressing::*;
+
+const BOND_NAMESPACE: &str = "bond namespace";
```
As a suggestion, because you are importing everything from `bond_common::addressing`, you can just get the bond namespace that is defined there. So you can use `FAMILY_NAMESPACE` to get the bond namespace and don't have to redefine it here. You might need to make FAMILY_NAMESPACE in `addressing` public.

shannynalayna commented:
@eloaverona @dplumb94 Updated, per both of your suggestions, including the Dockerfile

eloaverona commented on processor/src/handler.rs:
```diff
@@ -0,0 +1,58 @@
+        BondTransactionHandler {
+            family_name: "bond".to_string(),
+            family_versions: vec!["0.1".to_string()],
+            namespaces: vec![hash(&FAMILY_NAMESPACE, 6)],
```
No to be to picky.. But I was wondering if there should be a get_family_namespace_hash method or something like that in addressing? Just so we don't need this hard coded 6 here. If we are abstracting away all the addressing stuff anyway...

shannynalayna replied:
Yes, I initially had a prefix hash function that was basically the same as the hash function in the addressing file, which was consolidated into a single hash function after some review of that file. I can add back a prefix function which gets the bond namespace because I think, overall, that's a good approach. Especially if that namespace is being referenced in the processor.

shannynalayna replied:
@dplumb94 thoughts?

eloaverona replied:
Lols yeas.. I remember I suggested you to remove it xD

shannynalayna commented:
@eloaverona @dplumb94 modified the addressing functions

Commit:
Add namespace specific function for addressing

Adds function to create and get the bond namespace.
Allows for greater abstraction.

Signed-off-by: Shannyn Telander <telander@bitwise.io>
